### PR TITLE
Team Roles Doc - Captain and Technical Manager definitions

### PR DESCRIPTION
**Pull request by daviddias**
Here is a first lightweight definition of the so much misunderstood roles within the IPFS Working Groups.

I've deliberately made these lightweight to give a good base but at the same time invite everyone's definition and current understanding of these roles. 

All the Reviews are welcome!

### CODE REVIEW

File: TEAM_STRUCTURES.md [hunk 1]
```diff
@@ -78,12 +79,14 @@ The IPFS Project Working Group Community serves as the point of coordination for
...
 - Research, develop and build new ways to help async teams coordinate at scale.
-- Recognize needs of the project and develop solutions to meet them. 
+- Recognize needs of the project and develop solutions to meet them.
 
 ### JavaScript IPFS implementation
 
 - **Coordination**: https://github.com/ipfs/pm/blob/master/JS_CORE_DEV_MGMT.md
-- **Captain**: [David Dias](https://github.com/diasdavid/)
+- **Captain**: [Alan Shaw](https://github.com/alanshaw/)
+- **Technical Manager**: N/A
```
Comment (diasdavid) [inline TEAM_STRUCTURES.md]: @alanshaw I hope with this to finally pass the well deserved and recognized baton without dropping a bunch of Management in tasks on top of you ad-aeternum.

Comment (alanshaw) [reply TEAM_STRUCTURES.md]: 👍 woooo!

File: TEAM_ROLES.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# IPFS Team Roles
+
+The individual contributors to the IPFS Project and Org often find themselves wearing many hats while performing they day to day tasks or on the projects they are focused on. We enable this by design to make the IPFS Org very permeable and empower each team to find its perfect balance for the challenges they are facing.
```
Comment (alanshaw) [inline TEAM_ROLES.md]: "while performing they day" => "while performing their day"

File: TEAM_ROLES.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+This doc presents the definition of only two roles, these are a) the ones we identified as necessary to have a well functioning Working Group and b) the ones that raise more questions. There are many other roles (project manager, designer, UX researcher, developer, systems engineer, etc), that we are not defining in this doc (yet).
+
+## IPFS Working Groups specific roles
+
+There are mainly two that are unique to our [Team Structures](TEAM_STRUCTURES.md). The responsibilities within these roles can be shared between multiple individuals, however, they must be owned by a single person that takes the responsibility to ensure the work is happening.
+
+### Captain
+
+The Working Group Captain is a champion for the Product and/or Focus Area. It takes the lead on writting or guiding the conversation specs, documentation and other artifacts to support the team. The Captain is also the gatekeeper of the Working Group Roadmap and accumlator of the Working Group Knowledge, guiding the group to made good decisions.
```
Comment (alanshaw) [inline TEAM_ROLES.md]: typo "writting" => "writing"

File: TEAM_ROLES.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+### Technical Manager
+
+The Technical Manager (sometimes referenced as Project Manager, Technical Project Manager and cat herder) is a team enabler. It ownes the Quarterly Planning process including OKRs and Retrospectives. It ensures that the coordination strategy the WG selected is well executed (weekly syncs, taking notes, communicating needs to other WGs).
```
Comment (alanshaw) [inline TEAM_ROLES.md]: "It ownes", "It ensures" => "They own", "They ensure". Less object more person?

File: TEAM_ROLES.md (hunk 3 above)
Comment (alanshaw) [inline TEAM_ROLES.md]: "It takes the lead" => "They take the lead"?

File: TEAM_STRUCTURES.md (hunk 1 above)
Comment (alanshaw) [inline TEAM_STRUCTURES.md]: "N/A" => "TBC"?

**Comment by diasdavid** [comment]
Thank you for the CR, @alanshaw. @ipfs/working-group-captains any other comments? Good to merge?

**Comment by hsanjuan** [comment]
I think it's perfect to have these definitions in js-ipfs if you feel confortable with them, but why pushing them down to everyone even when in most WGs the role falls on top of the same person or directly doesn't apply?

IMHO the two roles are Project manager (okrs, roadmap, maybe product docs) and Tech lead (tech discussions, mentoring, development). But I don't want to start a org-wide-discussion on why this is better or worse, because I don't see the need to impose a definition to other WGs which may choose to handle responsibilities however it fits them. ¯\_(ツ)_/¯

**Comment by olizilla** [comment]
@hsanjuan I think this is the right place to have a discussion on what the roles are. Project manger & Tech lead are more familiar to me.

There is value in trying to co-ordinate on these definitions, particulaly if that leads to a clear way to demonstrate that captains are overloaded, and a path to fixing that. 

Perhaps we should aim to move away from "captain" as a role, as right now it means "tech lead who also does project management mostly because of history and the ratio of employees to endevours"

**Comment by diasdavid** [comment]
@hsanjuan when it comes to keeping things straight forward and letting Working Groups self manage to the best of their needs, [I am always the first one in the line to defend the benefits that this org has gotten thanks to that flexibility](https://drive.google.com/file/d/1xnssXqHdIK6C4BrwnVw7BQ9a6XgR86VG/view). 

That said, I also acknowledge the countless numbers of requests for clarifications of what are the Captain duties and understand the point points that lack of clarity leads to.

Since the Captain role was originally designed, the whole org evolved both in practices and size, which led it to be more structured and adding a pile of management tasks to a role that was previously a _tech lead_/_champion_ like role for a focus area. That lead of the results of that was that either Captains ignore that transition completely (leading to some teams floating without clear direction for multiple quarters) or they started getting completely DOS'ed by management tasks tasks).

That said part II, as @olizilla well said, let's have this discussion here.

On another note. The reason why I selected the role of`Technical Manager` vs the more generic `Project Manager` is that there is a lot value on being explicit about having managers that understand clearly the technology, the pain points of development and how to help dev teams be effective, vs. the classical "Project Manager" that covers the part of managing projects.

**Comment by momack2** [comment]
To me, a "techincal manager" sounds like a "people manager" - if the goal is for this position to own the process coordination, OKRs, meeting running, etc, then I think we should change this title to TPM (Technical Project Manager) to meet both the technical need while not straying into "people management" territory. Speaking of that, captains also frequently end up wearing the people management, coaching, and resourcing hats (if very lightly), so that's another role I think we should define.

You say the TPM is also the person handling cross-wg communication and alignment. In the few places where these aren't the same person, that role seems to actually be more on the captain's responsibilities.

To me, instead of defining a new top-level "position", I'd structure this guide around a few **classes** of ownership that captains and others frequently take on. One is "Steering the ship" (captain), another is "coordinating the process", a third is "coaching the people", etc. Then in most cases there's a single captain owning all these "roles" - but in some cases there might be other individuals stepping up to take on some other role. The way I'd imagine that changing our "team structures" doc is to have an individual listed with the set of roles they are owning instead of calling out a "same as captain" slot for TPM. This also enables other working groups to be a little more flexible in their own role setup.

**Comment by diasdavid** [comment]
I went back and forth with the Technical Manager vs Technical Project Manager. The reason why I went with Technical Manager in the end was so that for the Working Groups that have multiple projects and therefore multiple project managers for each project, we didn't fall into some weird confusion.

Keep comments and thinking coming, this is good! :)

**Comment by momack2** [comment]
To paint out the scenario - you're talking about a case where there are two products owned by a working group (say GUI WG owns product Companion and WebUI), and hypothetically each of those products could have a different TPM associated with them that is _also different_ from the TPM leading the wider GUI working group. That situation seems OK to me even if they're all "TPM"s... The TPM listed on the **WG team structure entry** would be the **WG TPM** - while Companion and WebUI TPMs would be listed separately in repos/docs about their specific products. This is similar to how there might be folks who "steer the ship" for a product within a working group, or people-coach just the people in that product - they could consider themselves "Companion Captain**" or "WebUI Coach". If we want to add products owned by WGs to this doc (ex adding Companion and WebUI as "products" under the GUI WG team structure), the product TPMs could be called out there along with the products' "ship steerers" and "people coaches". 

**If folks are concerned about using "Captain" for both products and WGs (I think it's fine, but folks might disagree) - we could rename doing the "technical ship steering" role to "tech lead" (or similar) to differentiate the "captain" title for working group lead maintainers from the 3 roles a captain might take on (tech steering, process guiding, people coaching). 

I'll create a PR to this PR with my proposed change. ;)

**Comment by olizilla** [comment]
@momack2 +1 to defining roles that 1 or multiple people can own.

We should check that we agree on the list of responsibilities that make up each role. Can we indentify the minimum set of things we are willing to say each team _must_ do, in order to play well with others and have it's actions be legible to the wider org? With that list defined, we can pin some role names on a collection of repsonsibilities as a short hand for saying who is responsible for what. The role name is less important than the set of things that must be done. 

Where we have cases where all the responsibilities are on one person, those teams should be first in the queue for hiring additional crew, if they decide they need help. For smaller teams there is a case for a time-share PMs, and designers who help multiple working groups.

More generally, I'm assuming that we want to optimise for managable team sizes. In cases where a team is currently in charge of multiple products, it should still only have 1 person in the PM role. If managing multiple products becomes too much, that would be the time to think about splitting it into 1 team per product. My ideal is many teams with (up to) 5 Dev + (at least) 1  Design + 1 PM people on them, and adjust the responsibilites of the team to fit their capacity, rather than adjusting the teams to fit projects whose scope chages over time. I realise that some amount flex in both directions is needed, but breaking large projects into areas of responsibility seems like our thing.

**Comment by meiqimichelle** [comment]
This is a great thread! I agree with the direction it seems to be trending -- that we should work on a list of responsibilities that is the minimum set of things we think *someone* should be doing for each team to, as @olizilla put it, `play well with others and have it's actions be legible to the wider org`. Once we work as a team to get all the Venn diagrams out of our individual understandings and into something we can play around in and move things around (Hark! [Check out this Mural](https://app.mural.co/invitation/mural/protocollabs6957/1542841118216?sender=michellehertzfeld1447&key=4df5ab83-919d-4dfb-a8ce-b75e1fb88212) I made for just this purpose!), I am sure the categories of work will bubble up.

**So, my ask: [please pop into that Mural](https://app.mural.co/invitation/mural/protocollabs6957/1542841118216?sender=michellehertzfeld1447&key=4df5ab83-919d-4dfb-a8ce-b75e1fb88212)** and follow the directions (and/or tell me if it doesn't make sense). I broadened the directions to "things you think every team should do" (not scoped to Oli's sentence) because I think it will be easier for all of us to do a BIG braindump, and then separate out the things that we think a P\*M\Captain\TPM\whatever should do. Otherwise, we might miss some things that, right now, some folks don't associate with P\*M, but others do, etc etc.

Other notes:

- +1 to the assumption that we should be optimizing for manageable team sizes, or, to state that another way -- balanced teams who have all the necessary roles covered in ways that are sustainable -- and that we should be putting unbalanced teams first in queue for hiring additional people. Depending on the project scope, that might translate to a different number of people or combination of skillset.

- As for people management (one-on-ones, career coaching), I believe that should be a track separate from projects entirely, as per https://github.com/protocol/LabOS/issues/36 . But for the time being, I put a few stickie notes in the Mural with these tasks so we can talk about them.

- As for ideal team size, very, very often when I worked staffing client design-and-build projects, the breakdown would be:

  - *Discovery phase*: 1 UX/PM, 1 user researcher (aka, don't build anything till you understand the problem, detailed use case, and human factors)
  - *Build-something-with-UI-based-on-discovery*: 1 UX/PM or TPM, 1 content designer or user researcher, 1 visual designer or front end designer, 1 front end engineer, 1 back end engineer
  - *Build-something-without-big-UI-based-on-discovery*: 1 UX/PM or TPM, 1 content designer or user researcher or front end designer, 3 engineers of various types, depending on project need

- And then, like what @olizilla said if I understood him correctly, for large projects you'd separate the concerns and build versions of these teams for each part of the project. In that case, you might have more dev-vs-design because the user research would be shared across several small teams.

**Comment by eefahy** [comment]
Is it possible to get to the mural without making an account?

**Comment by daviddias** [comment]
I would like to get this PR merged and have the information it offers available for all our contributors. I know that there are other tangent discussions happening on this thread and we can continue discussing them on issues.

Sounds good? Any objections?

**Pull request merged**